### PR TITLE
Added an eventDirection field in the vocabulary Event

### PR DESCRIPTION
**Pull request by chevtche**

### CODE REVIEW

File: zeq/detail/vocabulary.cpp [hunk 1]
```diff
@@ -41,6 +41,7 @@ Event serializeVocabulary( const EventDescriptors& vocabulary )
     std::vector< uint64_t > eventTypeLows;
     std::vector< uint64_t > eventTypeHighs;
     std::vector< flatbuffers::Offset<flatbuffers::String> > schemas;
+    std::vector< int8_t > eventDirections;
```
Comment (tribal-tec) [inline zeq/detail/vocabulary.cpp]: uint8_t?

File: zeq/detail/vocabulary.fbs [hunk 2]
```diff
@@ -3,13 +3,16 @@
 
 namespace zeq.vocabulary;
 
-//A struct containing restNames, eventHighs, eventLows, schemas would be more
-//clear but flatbuffers does not support strings in structures.
+enum EventDirection : byte { Subscriber = 0, Publisher, Both }
```
Comment (tribal-tec) [inline zeq/detail/vocabulary.fbs]: ubyte? Omit the =0? It's implicit according to documentation

File: zeq/eventDescriptor.h [hunk 3]
```diff
@@ -46,6 +58,9 @@ struct EventDescriptor : public boost::noncopyable
 
     /** @return the flatbuffers schema string*/
     ZEQ_API const std::string& getSchema() const;
+
+    /** @return the zeq event's direction (Subscribed, Pulished or both)*/
+    ZEQ_API const EventDirection& getEventDirection() const;
```
Comment (tribal-tec) [inline zeq/eventDescriptor.h]: no const ref for enums

File: zeq/eventDescriptor.h [hunk 4]
```diff
@@ -31,7 +42,8 @@ struct EventDescriptor : public boost::noncopyable
      */
     ZEQ_API EventDescriptor( const std::string& restName,
                              const uint128_t& eventType,
-                             const std::string& schema );
+                             const std::string& schema,
+                             const EventDirection& eventDirection);
```
Comment (tribal-tec) [inline zeq/eventDescriptor.h]: no const ref for enums

**Comment by chevtche** [comment]
Retest this please

**Comment by chevtche** [comment]
Retest this please

**Comment by chevtche** [comment]
Retest this please

**Comment by chevtche** [comment]
Retest this please

**Comment by chevtche** [comment]
Retest this please

File: zeq/detail/eventDescriptor.h [hunk 5]
```diff
@@ -18,19 +18,22 @@ namespace detail
...
 {
     EventDescriptor( const std::string& restName, const uint128_t& eventType,
-                     const std::string& schema )
+                     const std::string& schema, const EventDirection eventDirection )
         : _restName( restName )
         , _eventType( eventType )
         , _schema( schema )
+        , _eventDirection( eventDirection )
     {}
     const std::string& getRestName() const { return _restName; }
```
Comment (tribal-tec) [inline zeq/detail/eventDescriptor.h]: remove the functions and make the members public? They are const and it's a detail struct, so the getters are actually overkill :)

File: zeq/detail/eventDescriptor.h [hunk 6]
```diff
@@ -18,19 +18,16 @@ namespace detail
...
         , _eventType( eventType )
         , _schema( schema )
+        , _eventDirection( eventDirection )
     {}
-    const std::string& getRestName() const { return _restName; }
-    const uint128_t& getEventType() const { return _eventType; }
-    const std::string& getSchema() const { return _schema; }
-
-private:
     const std::string _restName;
```
Comment (tribal-tec) [inline zeq/detail/eventDescriptor.h]: remove the underscores, it's not private anymore

**Comment by tribal-tec** [comment]
If no other comments from somebody else, I'll merge after the fixed underscores.

**Pull request merged**